topy/topy.py

```diff
@@ -121,11 +121,27 @@ def sanitize_filename(filename):
 def print_diff(filename, old, new, stream=sys.stdout):
     """Diffs the `old` and `new` strings and prints as unified diff to file-like object `stream`."""
 
-    # TODO: color output for terminals
     lines = unified_diff(old.splitlines(True), new.splitlines(True), filename, filename)
+    lines = map(lambda l: determine_color(l, stream=stream), lines)
     stream.writelines(lines)
 
 
+def determine_color(line, stream):
+    """Determines the color of the diffs."""
+    return add_output_color(line) if is_color_output_required(stream) else line
+
+
+def is_color_output_required(stream):
+    """Determines whether to output the line diffs in color or not."""
+    return False if opts.color == "never" else True if opts.color == "always" else hasattr(stream, 'isatty') and stream.isatty()
```

Comment:
Personally I'd find both `is_color_output_required` and `add_output_color` easier to read if they used multiline `if/elif/else`

Reply:
👍 Will do